### PR TITLE
deprecate ready setting

### PR DESCRIPTION
**Pull request by JarbasAl**
companion to https://github.com/OpenVoiceOS/ovos-config/pull/166

NOTES:
- to check if a service is ready, this should be used https://github.com/OpenVoiceOS/ovos-utils/blob/dev/ovos_utils/process_utils.py#L124
- now that skills are not necessarily managed by core,  `mycroft.ready` message was emitted almost randomly and was not dependendable on
- `mycroft.ready` is moved to finished booting skill and no longer blocks loading of ovos-core
- external skills / anything registering intents should still use `mycroft.ready` in combination with ProcessUtils linked above to determine when to load

tests moved to other repos:
- https://github.com/OpenVoiceOS/ovos-config/pull/168
- https://github.com/OpenVoiceOS/ovos-bus-client/pull/125
- https://github.com/OpenVoiceOS/ovos-utils/pull/274

## Summary by CodeRabbit

- **New Features**
	- Enhanced skill management processes for improved efficiency.
	- Refined handling of skills requiring network or internet connections.

- **Bug Fixes**
	- Improved synchronization checks during skill loading.
	- Deprecation of outdated methods to streamline skill management.

- **Tests**
	- Expanded test coverage for blacklisted skill loading.
	- Adjustments made to the validation of event handlers in tests.
	- Removed obsolete test methods to enhance clarity.

### CODE REVIEW

**Comment by NeonDaniel** [comment]
Why deprecate this? It is used in many skills and integrations to determine when core services are ready for normal usage

**Comment by JarbasAl** [comment]
discussion here https://matrix.to/#/!EOygeDJPfJQOfNacqH:matrix.org/$cN3qQ_K_CdrMSYXUx3rDRUGs60gwX3Q_QlK3vzeEkIo?via=matrix.org&via=tchncs.de&via=mozilla.org

**Comment by JarbasAl** [comment]
> Why deprecate this? It is used in many skills and integrations to determine when core services are ready for normal usage

edited original post to include some extra info about this

**Comment by NeonDaniel** [comment]
> discussion here https://matrix.to/#/!EOygeDJPfJQOfNacqH:matrix.org/$cN3qQ_K_CdrMSYXUx3rDRUGs60gwX3Q_QlK3vzeEkIo?via=matrix.org&via=tchncs.de&via=mozilla.org

I have used it in Neon without changes for probably more than a year now; changes to default configuration could be an alternative solution compared to removing configuration support completely.

If the meaning of "mycroft.ready" is changing here, I would ask that this emit something less ambiguous (maybe "mycroft.intent_service.ready", or implement ProcessStatus directly for consistency with other modules that report state. `mycroft.ready` emit should be configurable (possibly with a deprecation warning) since the meaning of the event has changed; this would allow anyone wanting to restore backwards-compat. to emit that message with a plugin (i.e. the React GUI might want to wait for the GUI core service to start).

File: ovos_core/skill_manager.py [hunk 1]
```diff
@@ -469,86 +363,29 @@ def _load_plugin_skill(self, skill_id, skill_plugin):
...
 
         self.status.set_ready()
 
-        if self._gui_event.is_set() and self._connected_event.is_set():
-            LOG.info("Skills all loaded!")
-        elif not self._connected_event.is_set():
-            LOG.info("Offline Skills loaded, waiting for Internet to load more!")
-        elif not self._gui_event.is_set():
-            LOG.info("Skills loaded, waiting for GUI to load more!")
+        self.bus.emit(Message('mycroft.ready'))
```
Comment (NeonDaniel) [inline ovos_core/skill_manager.py:386]: The meaning of this message is changed here; can this emit a new event that is not ambiguous and put `mycroft.ready` behind a config flag to enable a dev to implement backwards-compat. via a plugin?

**Comment by JarbasAl** [comment]
moved message to skill https://github.com/OpenVoiceOS/skill-ovos-boot-finished/pull/15

File: ovos_core/skill_manager.py [hunk 2]
```diff
@@ -564,7 +429,7 @@ def run(self):
 
     def _load_on_network(self):
         """Load skills that require a network connection."""
-        if self._detected_installed_skills: # ensure we have skills is installed
+        if self._detected_installed_skills:  # ensure we have skills is installed
```
Comment (goldyfruit) [inline ovos_core/skill_manager.py:432]: typo in the comment?

**Pull request merged**